browser-extension/mv2/src/background/background.js

```diff
@@ -1442,29 +1241,28 @@ BG.Methods.saveExecutionCount = async function (rule) {
 };
 
 BG.Methods.getTabSession = (tabId, callback) => {
-  chrome.tabs.sendMessage(
-    tabId,
-    { action: RQ.CLIENT_MESSAGES.GET_TAB_SESSION },
-    { frameId: 0 },
-    callback
-  );
+  chrome.tabs.sendMessage(tabId, { action: RQ.CLIENT_MESSAGES.GET_TAB_SESSION }, { frameId: 0 }, callback);
 };
 
 BG.Methods.sendAppliedRuleDetailsToClient = async (rule, requestDetails) => {
   const { tabId } = requestDetails;
 
-  chrome.tabs.sendMessage(tabId, {
-    action: RQ.CLIENT_MESSAGES.UPDATE_APPLIED_RULE_ID,
-    ruleId: rule.id,
-  });
+  chrome.tabs.sendMessage(
+    tabId,
+    {
+      action: RQ.CLIENT_MESSAGES.UPDATE_APPLIED_RULE_ID,
+      ruleId: rule.id,
+    },
+    (isFirstExecution) => {
```

Comment:
Can we rather send full rule definition to client in the message and let the content script trigger the event similar to popup and devtool?
In that case, the action can be renamed to `NOTIFY_RULE_APPLIED`.